internal/service/logs/anomaly_detector.go

```diff
@@ -0,0 +1,299 @@
+		Attributes: map[string]schema.Attribute{
+			names.AttrARN: framework.ARNAttributeComputedOnly(),
+			"log_group_arn_list": schema.ListAttribute{
+				CustomType:  fwtypes.ListOfStringType,
+				ElementType: types.StringType,
+				Required:    true,
+			},
+			"anomaly_visibility_time": schema.Int64Attribute{
+				Optional: true,
+				Computed: true,
+				Validators: []validator.Int64{
+					int64validator.AtLeast(7),
+					int64validator.AtMost(90),
+				},
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.UseStateForUnknown(),
+				},
+			},
+			"detector_name": schema.StringAttribute{
+				Optional: true,
+			},
+			"evaluation_frequency": schema.StringAttribute{
+				Optional: true,
+				Validators: []validator.String{
+					enum.FrameworkValidate[awstypes.EvaluationFrequency](),
+				},
+			},
+			"filter_pattern": schema.StringAttribute{
+				Optional: true,
+			},
+			names.AttrEnabled: schema.BoolAttribute{
+				Required: true,
+			},
+			names.AttrKMSKeyID: schema.StringAttribute{
+				Optional: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			names.AttrTags:    tftags.TagsAttribute(),
+			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+		},
+	}
+}
+
+func (r *resourceAnomalyDetector) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	conn := r.Meta().LogsClient(ctx)
+
+	var plan resourceAnomalyDetectorData
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	in := &cloudwatchlogs.CreateLogAnomalyDetectorInput{
+		Tags: getTagsIn(ctx),
+	}
+	resp.Diagnostics.Append(flex.Expand(ctx, plan, in)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	out, err := conn.CreateLogAnomalyDetector(ctx, in)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.Logs, create.ErrActionCreating, ResNameAnomalyDetector, plan.ARN.String(), err),
+			err.Error(),
+		)
+		return
+	}
+	if out == nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.Logs, create.ErrActionCreating, ResNameAnomalyDetector, plan.ARN.String(), nil),
+			errors.New("empty output").Error(),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(flex.Flatten(ctx, out, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	plan.ARN = flex.StringToFramework(ctx, out.AnomalyDetectorArn)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (r *resourceAnomalyDetector) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	conn := r.Meta().LogsClient(ctx)
+
+	var state resourceAnomalyDetectorData
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	out, err := findLogAnomalyDetectorByARN(ctx, conn, state.ARN.ValueString())
+	if tfresource.NotFound(err) {
+		resp.State.RemoveResource(ctx)
```

Comment:
nit: Let's log a warning here that this will be removed

```suggestion
                 resp.Diagnostics.Append(fwdiag.NewResourceNotFoundWarningDiagnostic(err))
		resp.State.RemoveResource(ctx)
```